File: ios/counter.m
//
//  counter.m
//  sliderApp
//
//  Created by Lisbaldy de Jesus on 15/07/2024.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Counter, NSObject)

RCT_EXTERN_METHOD(increment)
RCT_EXTERN_METHOD(printCount:(RCTResponseSenderBlock)callback)

@end

File: ios/counter.swift
//
//  counter.swift
//  sliderApp
//
//  Created by Lisbaldy de Jesus on 14/07/2024.
//

import Foundation

@objc(Counter)
class Counter: NSObject {

    private var count = 0

    @objc
    func increment() {
      count += 1;
    }
    
    @objc
    func printCount(_ callback: RCTResponseSenderBlock) {
      callback([count]);
    }
  
    @objc
    func requiresMainQueueSetup() -> Bool {
      return true;
    }
  
    @objc
    func constantsToExport() -> [String: Any] {
      return ["initialCount": 0];
    }
    
}
